base/src/commonTest/kotlin/utils/DecimalTest.kt

```diff
@@ -45,9 +44,19 @@ class DecimalTest {
         assertEquals(0.333, 1.toDecimal().div(3.toDecimal(), 3, RoundingMode.RoundDown).toDouble())
         assertEquals(0.334, 1.toDecimal().div(3.toDecimal(), 3, RoundingMode.RoundUp).toDouble())
 
-        assertFailsWith(DecimalException::class) { 1.0.toDecimal() / 0.0.toDecimal() }
-        assertFailsWith(DecimalException::class) { 1.0.toDecimal().div(0.0.toDecimal(), 3) }
-        assertFailsWith(DecimalException::class) { 1.0.toDecimal().div(0.0.toDecimal(), 3, RoundingMode.RoundDown) }
+        assertEquals(Decimal.PositiveInfinity, 1.0.toDecimal() / 0.0.toDecimal())
+        assertEquals(Decimal.PositiveInfinity, 1.0.toDecimal().div(0.0.toDecimal(), 3))
+        assertEquals(Decimal.PositiveInfinity, 1.0.toDecimal().div(0.0.toDecimal(), 3, RoundingMode.RoundDown))
+
+        assertEquals(Decimal.NegativeInfinity, (-1.0).toDecimal() / 0.0.toDecimal())
+        assertEquals(Decimal.NegativeInfinity, (-1.0).toDecimal().div(0.0.toDecimal(), 3))
+        assertEquals(Decimal.NegativeInfinity, (-1.0).toDecimal().div(0.0.toDecimal(), 3, RoundingMode.RoundDown))
+
+        assertEquals(Decimal.NaN, 0.0.toDecimal() / 0.0.toDecimal())
+        assertEquals(Decimal.NaN, 0.0.toDecimal().div(0.0.toDecimal(), 3))
+        assertEquals(Decimal.NaN, 0.0.toDecimal().div(0.0.toDecimal(), 3, RoundingMode.RoundDown))
+
+        assertEquals(Decimal.NaN, "Test".toDecimal())
```

Comment:
please also add checks on potential precession loss eg
* close to max values 
* ops that cause the loss 
```
1.0 - 0.05 - 0.05 actually produces 0.8999999999999999
```

Reply:
Again, it should throw an exception here